flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java

```diff
@@ -67,6 +67,22 @@ static Map<String, Integer> getIndex(List<String> idList) {
             ));
     }
 
+    static List<Branch> getAllValidBranches(Network network) {
+        return network.getBranchStream()
+            .filter(NetworkUtil::isConnected)
+            .filter(NetworkUtil::isInMainSynchronousComponent) // Is connectedCompenent enough ?
```

Comment:
if you need to resolve this later, add "TODO:" to the beginning of the comment, it will be highlighted by sonar

Reply:
done